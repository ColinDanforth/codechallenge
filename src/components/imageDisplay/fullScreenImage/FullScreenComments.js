import React from 'react'
import PropTypes from 'prop-types'

const rootCommentDiv = {
  maxWidth: '500px',
  width: 'auto',
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  height: 'auto',
  marginTop: '3%',
  marginBottom: '3%',
  backgroundColor: 'RGBA(256,256,256,0.9)',
  border: 'inset'
}

const pStyleWrapped = {
  wordWrap: 'normal',
  maxWidth: '500px',
  width: 'auto',
}

const headerAndDateBlock = {
  display: 'flex',
  flexDirection:'row',
  justifyContent: 'space-between',
  alignItems: 'flex-start',
  marginLeft: '1%',
  marginRight: '1%',
}

const pDivStyle ={
  marginLeft: '2%',
  marginRight: '2%',
  width: 'auto',
}

const headerStyleWrapped = {
  wordWrap: 'normal',
  maxWidth: '500px',
  width: 'auto',
  fontWeight: 'bold',
}

const commentsTitle = {
  alignSelf: 'center',
  fontSize: '1.3em',
  fontWeight: 'bold',
}

const commentBlock = {
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  margin: '2%',
  border: 'outset',
  background: 'RGBA(256,256,256,1)'
}

const noCommentsMessage={
  margin: '3%',
  alignSelf: 'center',
}

class FullScreenComments extends React.Component{
  constructor(){
    super()
    this.state={
      comments: {},
    }

    this.commentsBlock = this.commentsBlock.bind(this)
  }

  componentDidUpdate(){
    if(
      this.props.thisImagesComments.hasOwnProperty('comments')
      && this.state.comments !== this.props.thisImagesComments
    ){
      this.setState({
        comments: this.props.thisImagesComments
      })
    }
  }

  commentsBlock(){
    if(this.state.comments.hasOwnProperty('comments') && this.state.comments.comments.length > 0) {
      return (this.state.comments.comments.map((comment, i) => {
        return (
          <div style={commentBlock} key={i}>
            <div style={headerAndDateBlock}>
              <p style={headerStyleWrapped}>{comment.user.username}</p>
              <p>{comment.created_at}</p>
            </div>
            <div style={pDivStyle}>
              <p style={pStyleWrapped}>{comment.body}</p>
            </div>
          </div>
        )
      }))
    } else {
      return <p style={noCommentsMessage}>No comments yet, be the first to comment on this Great Image!</p>
    }
  }

  render(){
    return(
      <div style={rootCommentDiv}>
        <p style={commentsTitle}>User Comments</p>
        {this.commentsBlock()}
      </div>
    )
  }
}

export default FullScreenComments

FullScreenComments.propTypes={
  thisImagesComments: PropTypes.object,
}
