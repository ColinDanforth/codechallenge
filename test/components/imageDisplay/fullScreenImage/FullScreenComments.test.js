import React from 'react';
import FullScreenComments from '../../../../src/components/imageDisplay/fullScreenImage/FullScreenComments';
import renderer from 'react-test-renderer';

test('FullScreenComments should display a group of comment elements showing the appropriate text from the testData object', () => {
  const testData={
    comments: [{
      user: {
        username: 'TestUser'
      },
      created_at: '2018-01-01',
      body: 'Some Comment String',
    },{
      user: {
        username: 'TestUser2'
      },
      created_at: '2018-01-02',
      body: 'Some Comment String 2',
    }],
  }

  const component = renderer.create(
    <FullScreenComments thisImagesComments={testData}/>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

})

test('FullScreenComments should show comment object with a string "No comments yet, be the first to comment on this Great Image!", empty comments array', () => {
  const testData={
    comments: [],
  }

  const component = renderer.create(
    <FullScreenComments thisImagesComments={testData}/>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

})

test('FullScreenComments should show comment object with a string "No comments yet, be the first to comment on this Great Image!", no comments key.', () => {
  const testData={
  }

  const component = renderer.create(
    <FullScreenComments thisImagesComments={testData}/>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

})
