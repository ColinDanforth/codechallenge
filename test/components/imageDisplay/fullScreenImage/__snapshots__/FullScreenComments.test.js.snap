// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullScreenComments should display a group of comment elements showing the appropriate text from the testData object 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "RGB(256,256,256)",
      "borderBottom": "inset",
      "display": "flex",
      "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
      "flexDirection": "column",
      "height": "auto",
      "justifyContent": "space-around",
      "maxWidth": "100%",
      "width": "auto",
    }
  }
>
  <p
    style={
      Object {
        "alignSelf": "center",
        "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
        "fontSize": "1.3em",
        "fontWeight": "bold",
      }
    }
  >
    User Comments
  </p>
  <div
    style={
      Object {
        "background": "RGBA(256,256,256,1)",
        "border": "outset",
        "display": "flex",
        "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
        "flexDirection": "column",
        "height": "auto",
        "justifyContent": "center",
        "margin": "2%",
        "maxWidth": "96%",
        "minWidth": "96%",
        "width": "auto",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "flex-start",
          "display": "flex",
          "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginLeft": "1%",
          "marginRight": "1%",
        }
      }
    >
      <p
        style={
          Object {
            "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
            "fontWeight": "bold",
            "maxWidth": "90%",
            "width": "auto",
            "wordWrap": "normal",
          }
        }
      >
        TestUser
      </p>
      <p>
        2018-01-01
      </p>
    </div>
    <div
      style={
        Object {
          "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
          "marginLeft": "2%",
          "marginRight": "2%",
          "width": "auto",
        }
      }
    >
      <p
        style={
          Object {
            "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
            "maxWidth": "100%",
            "width": "auto",
            "wordWrap": "normal",
          }
        }
      >
        Some Comment String
      </p>
    </div>
  </div>
  <div
    style={
      Object {
        "background": "RGBA(256,256,256,1)",
        "border": "outset",
        "display": "flex",
        "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
        "flexDirection": "column",
        "height": "auto",
        "justifyContent": "center",
        "margin": "2%",
        "maxWidth": "96%",
        "minWidth": "96%",
        "width": "auto",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "flex-start",
          "display": "flex",
          "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginLeft": "1%",
          "marginRight": "1%",
        }
      }
    >
      <p
        style={
          Object {
            "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
            "fontWeight": "bold",
            "maxWidth": "90%",
            "width": "auto",
            "wordWrap": "normal",
          }
        }
      >
        TestUser2
      </p>
      <p>
        2018-01-02
      </p>
    </div>
    <div
      style={
        Object {
          "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
          "marginLeft": "2%",
          "marginRight": "2%",
          "width": "auto",
        }
      }
    >
      <p
        style={
          Object {
            "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
            "maxWidth": "100%",
            "width": "auto",
            "wordWrap": "normal",
          }
        }
      >
        Some Comment String 2
      </p>
    </div>
  </div>
</div>
`;

exports[`FullScreenComments should show comment object with a string "No comments yet, be the first to comment on this Great Image!", empty comments array 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "RGB(256,256,256)",
      "borderBottom": "inset",
      "display": "flex",
      "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
      "flexDirection": "column",
      "height": "auto",
      "justifyContent": "space-around",
      "maxWidth": "100%",
      "width": "auto",
    }
  }
>
  <p
    style={
      Object {
        "alignSelf": "center",
        "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
        "fontSize": "1.3em",
        "fontWeight": "bold",
      }
    }
  >
    User Comments
  </p>
  <p
    style={
      Object {
        "alignSelf": "center",
        "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
        "margin": "3%",
      }
    }
  >
    No comments yet, be the first to comment on this Great Image!
  </p>
</div>
`;

exports[`FullScreenComments should show comment object with a string "No comments yet, be the first to comment on this Great Image!", no comments key. 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "RGB(256,256,256)",
      "borderBottom": "inset",
      "display": "flex",
      "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
      "flexDirection": "column",
      "height": "auto",
      "justifyContent": "space-around",
      "maxWidth": "100%",
      "width": "auto",
    }
  }
>
  <p
    style={
      Object {
        "alignSelf": "center",
        "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
        "fontSize": "1.3em",
        "fontWeight": "bold",
      }
    }
  >
    User Comments
  </p>
  <p
    style={
      Object {
        "alignSelf": "center",
        "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
        "margin": "3%",
      }
    }
  >
    No comments yet, be the first to comment on this Great Image!
  </p>
</div>
`;
